src/Build/BackEnd/Components/RequestBuilder/Lookup.cs

```diff
@@ -1072,11 +1072,7 @@ public void ApplyModifications(MetadataModifications other)
                 if (other._keepOnlySpecified)
                 {
                     // Any metadata not specified in other must be removed from this one.
-                    var metadataToRemove = new List<string>(_modifications.Keys.Where(m => other[m].Remove));
-                    foreach (var metadata in metadataToRemove)
-                    {
-                        _modifications.Remove(metadata);
-                    }
+                    _modifications = _modifications.Where(m => !other[m.Key].Remove).ToDictionary(x => x.Key, x => x.Value);
```

Comment:
This allocates a new copy of the dictionary, while the old was still using the same memory. Wouldn't this increase allocations?